// ngx-font-awesome/bundles/ngx-font-awesome.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-font-awesome', ['exports', '@angular/core', '@angular/common'], factory) :
    (global = global || self, factory(global['ngx-font-awesome'] = {}, global.ng.core, global.ng.common));
}(this, function (exports, core, common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxFontAwesomeComponent = /** @class */ (function () {
        function NgxFontAwesomeComponent() {
            this.classes = '';
        }
        /**
         * @return {?}
         */
        NgxFontAwesomeComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        /**
         * getClasses()
         * Function will return Options in class form
         */
        /**
         * getClasses()
         * Function will return Options in class form
         * @return {?}
         */
        NgxFontAwesomeComponent.prototype.getClasses = /**
         * getClasses()
         * Function will return Options in class form
         * @return {?}
         */
        function () {
            (this.size) ? this.classes += "fa-" + this.size + " " : '';
            (this.rotate) ? this.classes += "fa-rotate-" + this.rotate + " " : '';
            (this.flip) ? this.classes += "fa-flip-" + this.flip + " " : '';
            (this.customClass) ? this.classes += this.customClass + " " : '';
            return this.classes;
        };
        NgxFontAwesomeComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-fa',
                        template: "\n    <i class=\"fa fa-{{ name }}\"\n      [ngClass]=\"getClasses()\"\n      [class.fa-fw]=\"fixed\"\n      [class.fa-border]=\"border\"\n      [class.fa-spin]=\"animation\"\n      aria-hidden=\"true\">\n    </i>\n  ",
                        changeDetection: core.ChangeDetectionStrategy.OnPush
                    }] }
        ];
        /** @nocollapse */
        NgxFontAwesomeComponent.ctorParameters = function () { return []; };
        NgxFontAwesomeComponent.propDecorators = {
            name: [{ type: core.Input }],
            size: [{ type: core.Input }],
            animation: [{ type: core.Input }],
            rotate: [{ type: core.Input }],
            flip: [{ type: core.Input }],
            fixed: [{ type: core.Input }],
            border: [{ type: core.Input }],
            customClass: [{ type: core.Input }]
        };
        return NgxFontAwesomeComponent;
    }());
    if (false) {
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.name;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.size;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.animation;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.rotate;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.flip;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.fixed;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.border;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.customClass;
        /** @type {?} */
        NgxFontAwesomeComponent.prototype.classes;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxFontAwesomeModule = /** @class */ (function () {
        function NgxFontAwesomeModule() {
        }
        NgxFontAwesomeModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: [NgxFontAwesomeComponent],
                        imports: [
                            common.CommonModule
                        ],
                        exports: [NgxFontAwesomeComponent]
                    },] }
        ];
        return NgxFontAwesomeModule;
    }());

    exports.NgxFontAwesomeModule = NgxFontAwesomeModule;
    exports.ɵa = NgxFontAwesomeComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
